GameOfThrones/Controllers/HouseController.cs

```diff
@@ -0,0 +1,39 @@
+﻿using GameOfThrones.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+
+namespace GameOfThrones.Controllers
+{
+    public class HouseController : Controller
+    {
+        private CastleContext db = new CastleContext();
+
+        // GET: House
+        public ActionResult Index()
+        {
+            return View(db.Houses.ToList());
+        }
+
+        public ActionResult Characters(int? id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(System.Net.HttpStatusCode.BadRequest);
+            }
```

Comment:
Interesting use of Http status codes as exceptions if behaviour is not valid.
In this case, would returning a HTTP code be over-the-top for this scenario - would it be better to rather than fail the app entirely, return some text or a warning back to the webpage instead?

Reply:
I thought that was quite an extreme thing to do too. Now it says that no characters have been found which allows the user to navigate to another page or alter their query/URL